agent/app/agent_capability.go

```diff
@@ -275,13 +275,13 @@ func (agent *ecsAgent) appendTaskCPUMemLimitCapabilities(capabilities []*ecs.Att
 	if agent.cfg.TaskCPUMemLimit.Enabled() {
 		if _, ok := supportedVersions[dockerclient.Version_1_22]; ok {
 			capabilities = appendNameOnlyAttribute(capabilities, attributePrefix+capabilityTaskCPUMemLimit)
-		} else if agent.cfg.TaskCPUMemLimit == config.ExplicitlyEnabled {
+		} else if agent.cfg.TaskCPUMemLimit.Value == config.ExplicitlyEnabled {
```

Comment:
my thought was that we could have an interface (`config.Boolean`) that doesn't return whether it's explicitly enabled or disabled, just whether it's enabled/disabled and set/unset. That way this if-statement would change to something like this:

```
else if agent.cfg.TaskCPUMemLimit.Enabled() && !agent.cfg.TaskCPUMemLimit.Unset() {
```

Reply:
Interface itself isn't a problem, but from my trying to implement this I found that agent's config has fair bit of built in assumptions that got me into few rabbit holes (e.g. non-nil config values are expected throughout tests and code).